File: src/main/Java/com/futureclient/modules/render/Fullbright.java
package com.futureclient.modules.render;

import com.futureclient.api.Category;
import com.futureclient.api.Module;

public class Fullbright extends Module {
    private float oldGamma;

    public Fullbright() {
        super("Fullbright", Category.RENDER, 0);
    }

    @Override
    public void onEnable() {
        oldGamma = mc.gameSettings.gammaSetting;
        mc.gameSettings.gammaSetting = 100f;
    }

    @Override
    public void onDisable() {
        mc.gameSettings.gammaSetting = oldGamma;
    }
}

File: src/main/kotlin/com/futureclient/impl/modules/render/Fullbright.kt
package com.futureclient.impl.modules.render

import com.futureclient.api.module.Category
import com.futureclient.api.module.Module
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
import net.minecraftforge.fml.common.gameevent.TickEvent

class Fullbright : Module("Fullbright", Category.RENDER) {
    private var originalGamma = 1f

    override fun onEnable() {
        super.onEnable()
        originalGamma = mc.gameSettings.gammaSetting
        mc.gameSettings.gammaSetting = 100f
    }

    override fun onDisable() {
        super.onDisable()
        mc.gameSettings.gammaSetting = originalGamma
    }
    
    // Periodically enforce it in case settings menu resets it
    @SubscribeEvent
    fun onTick(event: TickEvent.ClientTickEvent) {
        if (toggled && mc.gameSettings.gammaSetting != 100f) {
            mc.gameSettings.gammaSetting = 100f
        }
    }
}
